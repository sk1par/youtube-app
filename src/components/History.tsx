import React from 'react'
import { useDispatch, useSelector } from 'react-redux'
import selectedVideo from '../actions/selectedVideoActions';
import IVideos from '../models/videos.interface';
import { selectHistoryVideos } from '../selectors/selectors';

const History = () => {
    const historyVideos = useSelector(selectHistoryVideos);
    const dispatch = useDispatch();

    const selectVideo = (video: IVideos) => {
        dispatch(selectedVideo(video));
    }

    return (
        <>
        <div className='button' data-auto-id="history-title">
            <h1>History</h1>
        </div>

        <div>
            {historyVideos?.length > 0 ? historyVideos.map((video: IVideos) => {
               return  <p onClick={() => selectVideo(video)} className='link' key={video.id.videoId}
               data-auto-id='item'>{video.snippet.title}</p>
            }) : <p className='button' data-auto-id="no-results">No recent history</p>}
        </div>
        </>
    )
}

export default History
