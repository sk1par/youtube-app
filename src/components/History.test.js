import { useSelector } from 'react-redux';
import History from './History';
import Enzyme, { shallow } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { shallowToJson } from 'enzyme-to-json';
import selectedVideo from '../actions/selectedVideoActions';

Enzyme.configure({ adapter: new Adapter() });
jest.mock('react-redux', () => ({
  useDispatch: () => (fn) => () => fn(),
  useSelector: jest.fn(() => []),
}));

jest.mock('../actions/selectedVideoActions', () => {
  const selectedVideo = jest.fn();
  return selectedVideo;
});

const getVideo = () => ({
  id: {
    videoId: '1',
  },
  snippet: {
    description: 'test',
    title: 'test',
    thumbnails: {
      medium: {
        url: 'test',
      },
    },
  },
});
it('should render correctly', () => {
  const wrapper = shallow(<History />);
  const title = wrapper.find('[data-auto-id="history-title"]');
  expect(title).toHaveLength(1);
  const items = wrapper.find('[data-auto-id="item"]');
  expect(items).toHaveLength(0);
  expect(shallowToJson(wrapper)).toMatchSnapshot();
});

it('should render multiple items', () => {
  useSelector.mockReturnValueOnce([getVideo(), getVideo()]);
  const wrapper = shallow(<History />);
  const items = wrapper.find('[data-auto-id="item"]');
  expect(items).toHaveLength(2);
  expect(shallowToJson(wrapper)).toMatchSnapshot();
});

it('should receive obj after click', () => {
  const v = getVideo();
  useSelector.mockReturnValueOnce([v]);
  const wrapper = shallow(<History />);
  const item = wrapper.find('[data-auto-id="item"]');
  item.simulate('click');
  expect(selectedVideo).toHaveBeenCalledTimes(1);
  expect(selectedVideo).toHaveBeenCalledWith(v);
  expect(shallowToJson(wrapper)).toMatchSnapshot();
});

it('should display no results content', () => {
  const wrapper = shallow(<History />);
  const item = wrapper.find('[data-auto-id="no-results"]');
  expect(item).toHaveLength(1);
  expect(shallowToJson(wrapper)).toMatchSnapshot();
});