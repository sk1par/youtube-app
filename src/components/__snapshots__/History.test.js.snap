// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should display no results content 1`] = `
<Fragment>
  <div
    className="button"
    data-auto-id="history-title"
  >
    <h1>
      History
    </h1>
  </div>
  <div>
    <p
      className="button"
      data-auto-id="no-results"
    >
      No recent history
    </p>
  </div>
</Fragment>
`;

exports[`should receive obj after click 1`] = `
<Fragment>
  <div
    className="button"
    data-auto-id="history-title"
  >
    <h1>
      History
    </h1>
  </div>
  <div>
    <p
      className="link"
      data-auto-id="item"
      key="1"
      onClick={[Function]}
    >
      test
    </p>
  </div>
</Fragment>
`;

exports[`should render correctly 1`] = `
<Fragment>
  <div
    className="button"
    data-auto-id="history-title"
  >
    <h1>
      History
    </h1>
  </div>
  <div>
    <p
      className="button"
      data-auto-id="no-results"
    >
      No recent history
    </p>
  </div>
</Fragment>
`;

exports[`should render multiple items 1`] = `
<Fragment>
  <div
    className="button"
    data-auto-id="history-title"
  >
    <h1>
      History
    </h1>
  </div>
  <div>
    <p
      className="link"
      data-auto-id="item"
      key="1"
      onClick={[Function]}
    >
      test
    </p>
    <p
      className="link"
      data-auto-id="item"
      key="1"
      onClick={[Function]}
    >
      test
    </p>
  </div>
</Fragment>
`;
